doc/source/file_formats.rst

```diff
@@ -1,23 +1,375 @@
+
+Note: the link should have a name as overall content, to be consistent
+with moleculetype and modification (and to increase clarity), and the
+filters should probably fall under a subsection [ filters ].
+
+Patterns
+^^^^^^^^
+
+Patterns is a subsection that lists patterns of atoms to which the
+link applies. Each line in the subsection describes a pattern. One of
+the patterns must apply for the link to match. A pattern consists of
+atom identifiers. Each atom identifier consists of a name which may be
+preceded by a shift and which may be further specified using an
+attributes statement. The shift indicates a relative position in terms
+of residues:
+
+* +, ++, +++ : 	first, second, third following residue
+* -, --, --- : 	first, second, third previous residue
+* > :		other residue (not in order)
+
+Thus, +CA in amino acids refers to the C-alpha atom in the C-terminal
+connected neighbor, while >SG in the construction of a disulphide
+bridge will refer to the SG atom in the partner cysteine.
+
+An attributes statement is a JSON style mapping of key/value pairs,
+similar to those described above (see #meta). For a pattern to match,
+the atom names and the attributes need to match.
+
+Shift operators can also be used in the interaction subsections.
+
+Features
+^^^^^^^^
+
+The features subsection lists features to apply to the link
+itself. These can be used to control the application of links during
+the building of topologies. For example, setting the feature 'scfix'
+will cause the links to be applied only if the option -scfix is given
+to martinize2.
+
+Molmeta
+^^^^^^^
+
+The molmeta subsection lists metadata to be added/changed in the
+molecular graph. These metadata can be used (and modified) by
+Vermouth's processors and for provenance.
+
+Edges and non-edges
+^^^^^^^^^^^^^^^^^^^
+
+Within the context of links, the edges and non-edges specify that an
+edge should or shouldn't be present, respectively, for the link
+pattern to match. This behavior will likely be deprecated in the near
+future in favor of filter keywords 'edge' and 'nonedge'. The current
+behavior is really weird, because there is filtering and there are
+patterns to do matching, and then edges and non-edges also filter?
+Intuitively, edges and non-edges should specify the state after
+application of the link, similar to the other interactions. I would
+think that selection properties 'edge' and 'nonedge' in the filtering
+would be more intuitive and useful.
+
+Modifications
+^^^^^^^^^^^^^
+
+Modifications can be used to edit molecules or parts thereof (blocks),
+e.g., for specifying protonation states. Each modification starts with
+a line with the name. Thereafter may follow subsections as under
+links. A modification may add, remove, or change atoms, interactions
+and/or edges, using the corresponding subsections. Subsections
+patterns, features, molmeta, edges, and non-edges function as under
+links.
```

Comment:
this should probably be true, but currently isn't. I know for a fact that modifications currently can't remove interactions (and they definitely need to be able to!)